.github/workflows/test.yaml

```
@@ -0,0 +1,35 @@
+name: "test"
+
+on:
+  push:
+    branches: [main]
+  pull_request: {}
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  checks:
+    name: tests, fmt and clippy
```

Minor but prefer putting these in the order the steps run

```suggestion
    name: fmt, clippy and tests
```